### DISCUSSION TITLE
Unable to use APIs on Self-Hosted Airbyte instance

### DISCUSSION THREAD

**Discussion by Sahith-8055** [airbytehq/airbyte#41607] [Platform API Questions]
I have an airbyte instance which is hosted on kubernetes and fitted with basic authentication. I'm trying to use the APIs available at [Airbyte API](https://reference.airbyte.com/reference/getting-started) specifically the endpoint about retrieving the [List of Workspaces](https://reference.airbyte.com/reference/listworkspaces).

I had tried using Postman to send the request which returns me the below content -

```<!DOCTYPE html>
<html lang="en">
<head>
	<meta charset="utf-8">
	<link rel="icon" href="/favicon.ico">
	<meta name="viewport" content="width=device-width, initial-scale=1">
	<meta name="theme-color" content="#000000">
	<meta name="description"
		content="Airbyte is the turnkey open-source data integration platform that syncs data from applications, APIs and databases to warehouses.">
	<meta name="airbyte:sec-token" content="c82db11b-64df-413d-aba8-ee66b99c046f">
	<meta name="airbyte:version" content="0.50.7">
	<link rel="apple-touch-icon" href="/apple-touch-icon.png">
	<link rel="manifest" href="/manifest.json">
	<link rel="stylesheet" href="/index.css">
	<title>Airbyte</title>
	<script type="module" crossorigin src="/assets/index-b02d3f4a.js"></script>
	<link rel="stylesheet" href="/assets/index-61525523.css">
	<script data-script="config-overwrites">
		window.TRACKING_STRATEGY = "segment";
	</script>
</head>

<body>
	<noscript>
		<img src="/logo.png" alt="" width="90" height="94">
		<div class="card">You need to enable JavaScript to run this app.</div>
	</noscript>
	<div id="root"></div>
</body>
</html>
```

I tried using the cURL to send the request but I get the same response. Below is the way I had constructed the URL to communicate with the endpoints -

`https://<base-url>/v1/workspaces/` where `base-url` refers to the URL of the self-hosted instance.

<img width="897" alt="Screenshot 2024-07-10 at 11 40 11 PM" src="https://github.com/airbytehq/airbyte/assets/41897310/20c73f63-cd71-4d87-9b02-956a3061c00d">

Does someone let me understand what am I missing? Thanks for the support.

**Comment by NAjustin** [comment]
@Sahith-8055  You need to update your base path to `/api/public/v1/`. Previously there were two different deployments that ran; `server` (which hosted the config API, but also responded to the Airbyte API) and `api-server` (which proxied API requests to the server). They're [deprecating the stand-alone API server](https://reference.airbyte.com/reference/standalone-server-deprecation-and-migration-to-airbyte-server), so all API usage will point at the server.

It looks like you currently have half that change (you're talking to the server deployment), but you're missing the part where the path prefix changed.

**Reply by Sahith-8055** [reply]
Thanks for the reply @NAjustin. Could you point me to the location where I could find this piece of information as I don't see the same inside the API Docs?

**Reply by NAjustin** [reply]
Most of the official API docs are much more centered around Cloud and Self-hosted Enterprise and use those example paths, but in the [docs I linked before](https://reference.airbyte.com/reference/standalone-server-deprecation-and-migration-to-airbyte-server) it mentions the migration:
>For example, an API call `localhost:8006/v1/workspaces` must be migrated to: `localhost:8001/api/public/v1/workspaces`

Obviously that's talking about the default ports, but you can see the basic gist there.

Other cases where the OSS paths are mentioned (like [Authentication docs](https://reference.airbyte.com/reference/authentication) or [Terraform](https://reference.airbyte.com/reference/using-the-terraform-provider)) seem to still have the old paths/ports.

**Reply by NAjustin** [reply]
@marcosmarxm flagging this thread for you since it seems to be causing a lot of confusion in the community lately, from the [Getting Started](https://reference.airbyte.com/reference/getting-started) not mentioning OSS and talking about API keys (which don't apply to OSS) to the docs linked above for Terraform and such—might need a little docs pass to clear up the confusion. Even the standalone api-server deprecation notice mentions June 19 and July 19 for the deprecation date 🙃

**Reply by Sahith-8055** [reply]
Thanks @NAjustin for pointing it out. Still I'm unable to get the expected response. This time it returns **Object not found.** with a `404` code. I had missed providing this piece of information in my initial question that the **version of Airbyte UI** that I'm currently using is `0.50.7`. 

Does having a version different from the one mentioned in the **Migration Guide** affects the way we use the APIs? Can you let me know what should be the way forward? Thanks for the support.

**Reply by NAjustin** [reply]
@Sahith-8055 Ah, in that case, your version is prior to that change.

You might want to try instead connecting to the old path (without `/api/`) and connecting to the API server (usually on port `8006`) instead (which you may have to expose depending on your setup)—just keep in mind that will be changing in the coming days.

You could also look at upgrading your Airbyte version if you're able to, as there are a lot of fixes and improvements since the version you're on.

[a comment by an automated account is not shown]
